src/System.Net.Primitives/tests/FunctionalTests/CookieTest.cs

```diff
@@ -337,5 +337,26 @@ public static void ToString_Compare_Success()
             c.Port = "";
             Assert.Equal("name=value; $Path=path; $Domain=domain; $Port", c.ToString());
         }
+        
+        [Theory]
+        [MemberData(nameof(HashCodesTestData))]
+        public static void HashCodes_ShouldWorkLikeTuple(string name, string value, string path, string domain, int version)
```

Comment:
+1
We should have two tests: one for the common code (there's a common test project) and one that exercises the hashcode for Cookie. The expected hash code can be hard-coded instead of creating the impression that we'll always be compatible with what Tuple does.